torch/csrc/jit/codegen/cuda/runtime/block_reduction.cu

```diff
@@ -32,7 +32,10 @@ __device__ void blockReduce(
     T init_val) {
   unsigned int reduction_size = (X_REDUCE ? block_dim.x : 1) *
       (Y_REDUCE ? block_dim.y : 1) * (Z_REDUCE ? block_dim.z : 1);
-
+  if (reduction_size == 1) {
+    out = inp_val;
+    return;
+  }
```

Comment:
When `out` is on global memory, ignoring `read_write_pred` may not be always safe.